webui/react/src/App.tsx

```diff
@@ -89,19 +90,40 @@ const AppView: React.FC = () => {
   // Correct the viewport height size when window resize occurs.
   useLayoutEffect(() => correctViewportHeight(), [ resize ]);
 
+  const [ isAlive, setIsAlive ] = useState<boolean>(true);
+
+  useEffect(() => {
+    checkServerAlive().then(aIsAlive => {
+      if (!aIsAlive) {
+        setIsAlive(false);
+      }
+    });
+  }, []);
+
+  const UnreachableServerMessage = (
+    <PageMessage title="Server Unreachable">
+      <p>We cannot communicate with the server at {'"'}{serverAddress()}{'"'}.
```

Comment:
**nit:** small non-blocking suggestions
- `&quot;` instead of `{'"'}`. Just a guess but maybe slight benefit of not requiring extra JSX processing?
- `Unable to communicate` instead of `We cannot communicate`